### ISSUE TITLE
Unable to delete log file

### ISSUE THREAD

**Issue by shankhadevpadam** [opcodesio/log-viewer#198]
After upgrading to the latest version it is unable to delete the log file.

**Comment by arukompas** [comment]
hey @shankhadevpadam , I've just tested v2.0.1 and file deletion works as expected. Can you explain more about how you cannot delete logs? Is there an error? Console error?

Have you defined any Gates protecting against certain logs being deleted, like described here:
https://log-viewer.opcodes.io/docs/2.x/configuration/access-to-log-files#authorizing-log-file-deletion

**Comment by shankhadevpadam** [comment]
@arukompas 

When adding these inside AuthServiceProvider delete option is not visible in the file list.
```php
Gate::define('deleteLogFile', function (User $user, LogFile $file) {
    return $user->hasRole('Super Admin') ? true : false;
});

Gate::define('deleteLogFolder', function (User $user, LogFolder $folder) {
    return $user->hasRole('Super Admin') ? true : false;
});
```
But when I removed this gate from the AuthServiceProvider then the delete option is visible in the file list. But unable to delete the file.
```php
This action is unauthorized.
```

I used spatie role and permission for authentication. But I give full permission for the super admin role.

```php
Gate::before(function (User $user, $ability) {
    return $user->hasRole('Super Admin') ? true : null;
});
```

**Comment by arukompas** [comment]
@shankhadevpadam hm, interesting. It feels like during the API requests the user is not authenticated, thus does not pass any gates.

Can you try adding this middleware to the `log-viewer.api_middleware` config option?
```
\Laravel\Sanctum\Http\Middleware\EnsureFrontendRequestsAreStateful::class,
```

**Comment by shankhadevpadam** [comment]
@arukompas Thanks now working. This is a missing part of the documentation.

**Comment by GC-Mark** [comment]
We are also seeing this issue.

Do i really need to install Laravel Sanctum to so I can add that middleware to the config, or is there another way

<img width="954" alt="Screenshot 2023-02-23 at 15 22 36" src="https://user-images.githubusercontent.com/1477806/220950308-693dfc44-f479-4e8f-921c-a368e7a2b0b1.png">

**Comment by GC-Mark** [comment]
We don't actually have any Gates defined for deleting files, but even when i do define a Gate, the user is `null`

**Comment by arukompas** [comment]
No, that shouldn't be the case, of course. I'll look into it!

I'm trying to replicate the error you showed @GC-Mark , can you share what environment are you on and what's your app's setup like?
- PHP version
- Laravel version
- have you modified any HTTP Kernel middleware lists?
- have you applied any additional middlewares to Log Viewer?
- are you running over HTTPS or just HTTP ?

**Comment by arukompas** [comment]
Actually I found a way to reproduce. Working on a fix!

**Comment by arukompas** [comment]
Hey @GC-Mark @shankhadevpadam ,

I've made sure that Log Viewer works and API requests are authenticated correctly to be used with Auth gates.

**Important:** if you have previously published the `config/log-viewer.php` config file, please update it as per this changeset:
https://github.com/opcodesio/log-viewer/commit/533f0aef9a9f05f4b0c9e93d313b2f86545fec14#diff-ce45f607c40570c64d5225b23e190bc353063d8313fa422c52dd9bc95b684b7d

I'll update the docs later today. Thanks for your bug report!

**Comment by GC-Mark** [comment]
I can confirm, after updating to `v2.1.0` and updating the config file as per the commit, everything is now working as expected.

Thanks @arukompas 👍

**Comment by RogierVC** [comment]
I hate to comment on an old issue, but I got the same problem. I'm using a fresh install of version `3.17.1` and only have to following gate defined:

```
Gate::define('viewLogViewer', function (?User $user) {
    return $user ? true : false;
});
```

Adding a gate for `deleteLogFile` does not change a thing.

Viewing the log works fine. Also deleting multiple logs and folders is working, but deleting a single log file returns `403 Forbidden`.
I've analyzed the request headers and they are the same for these different requests.
